x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportCloseJobAction.java

```diff
@@ -249,21 +250,97 @@ void stopDatafeedsIfNecessary(OpenAndClosingIds jobIds, boolean isForce, TimeVal
                     if (runningDatafeedIds.isEmpty()) {
                         listener.onResponse(false);
                     } else {
-                        StopDatafeedAction.Request request = new StopDatafeedAction.Request(String.join(",", runningDatafeedIds));
-                        request.setForce(isForce);
-                        request.setStopTimeout(timeout);
-                        ClientHelper.executeAsyncWithOrigin(
-                            client,
-                            ClientHelper.ML_ORIGIN,
-                            StopDatafeedAction.INSTANCE,
-                            request,
-                            ActionListener.wrap(r -> listener.onResponse(r.isStopped()), listener::onFailure));
+                        if (isForce) {
+                            // A datafeed with an end time will gracefully close its job when it stops even if it was force stopped.
```

Comment:
> A datafeed with an end time will gracefully close its job when it stops even if it was force stopped.

This feels like a weird bug to me to begin with. It seems like a weird artifact of having them separate concepts.

Nothing to change, just an observation.

Reply:
Yes, I think this was unintentional.  The interactions between jobs and datafeeds are so complex that nobody understood the effect of having the datafeed close the job during its stop sequence.  If we'd realised the interactions we'd have probably made it so that a force-stopped datafeed with no end time force-closed its associated job.  However, that would be a breaking change now.  We should consider doing it for 8.0 _if_ the job/datafeed combination project hasn't got as far as combining the persistent tasks by then.